go-controller/pkg/ovn/pods.go

```diff
@@ -291,24 +290,22 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) (err error) {
 	lsp := &nbdb.LogicalSwitchPort{Name: portName}
 	if len(getLSP.UUID) == 0 {
 		lsp.UUID = libovsdbops.BuildNamedUUID()
-		// Only set ifaceIDVer for new LSP for correct ovn-kube upgrade, because for old OVS Interfaces
-		// iface-id-ver is not set => ovn-controller won't bind OVS Interface
-		ifaceIDVer = string(pod.UID)
 	} else {
 		lsp.UUID = getLSP.UUID
-		ifaceIDVer = getLSP.Options["iface-id-ver"]
 	}
 
 	lsp.Options = make(map[string]string)
-
 	// Unique identifier to distinguish interfaces for recreated pods, also set by ovnkube-node
 	// ovn-controller will claim the OVS interface only if external_ids:iface-id
 	// matches with the Port_Binding.logical_port and external_ids:iface-id-ver matches
 	// with the Port_Binding.options:iface-id-ver. This is not mandatory.
 	// If Port_binding.options:iface-id-ver is not set, then OVS
 	// Interface.external_ids:iface-id-ver if set is ignored.
-	if len(ifaceIDVer) != 0 {
-		lsp.Options["iface-id-ver"] = ifaceIDVer
+	// Only set iface-id-ver for new LSP for correct ovn-kube upgrade, because for old OVS Interfaces
+	// iface-id-ver is not set => ovn-controller won't bind OVS Interface.
+	// Update value for existing LSP if it was set before
+	if len(getLSP.UUID) == 0 || len(getLSP.Options["iface-id-ver"]) != 0 {
```

Comment:
So when we are upgrade from a version where the LSP didn't have `iface-id-ver` set, the condition would be `len(getLSP.UUID) != 0 && len(getLSP.Options["iface-id-ver"]) == 0` so we won't update the options.

Q: Why aren't we filling in this field for older pods as well post upgrade? Does it break the way ovn-controller works? Or I guess this works downstream since we have versioning but here there is no guarantee which ovn version we are using? I remember we discussed this during the original PR, just forgot.

Reply:
@npinaeva and I talked, and now I remember,

its because data plane has no idea that we've done a change in the control-plane during upgrades.

Reply:
yes thanks @npinaeva :)